nilearn/decoding/decoder.py

```diff
@@ -1298,11 +1311,34 @@ def __init__(
             n_jobs=n_jobs,
         )
 
+    def _more_tags(self):
+        """Return estimator tags.
+
+        TODO remove when bumping sklearn_version > 1.5
+        """
+        return self.__sklearn_tags__()
+
     def __sklearn_tags__(self):
+        """Return estimator tags.
+
+        See the sklearn documentation for more details on tags
+        https://scikit-learn.org/1.6/developers/develop.html#estimator-tags
+        """
+        # TODO
+        # get rid of if block
+        # bumping sklearn_version > 1.5
+        # see https://github.com/scikit-learn/scikit-learn/pull/29677
+        if SKLEARN_LT_1_6:
+            from nilearn._utils.tags import tags
+
+            return tags(require_y=True, niimg_like=True, surf_img=True)
+
+        from nilearn._utils.tags import InputTags
+
         tags = super().__sklearn_tags__()
-        ver = parse(sklearn_version)
-        if ver.release[1] >= 6:
-            tags.estimator_type = "classifier"
+        tags.estimator_type = "classifier"
+        tags.target_tags.required = True
+        tags.input_tags = InputTags(niimg_like=True, surf_img=True)
```

Comment:
Maybe I don't fully understand how these tag-related methods work, but do we still need to redefine them here if they are already in `_BaseDecoder` (which the Decoder object inherits)?